enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,618 @@
+* Identify and integrate with a technology or partner that can provide storage in a two-node environment.
+
+### Non-Goals
+
+* Reliance on traditional third-node or SNO setups.
+* Make sure we don't prevent upgrade/downgrade paths between two-node and traditional architectures
+* Adding worker nodes
+* Support for platforms other than bare metal including automated ci testing
+* Support for other topologies (eg. hypershift)
+* Failover time: if the leading node goes down, the remaining nodes takes over and gains operational state (writable) in less than 60s
+* support full recovery of the workload when the node comes back online after restoration - total time under 15 minutes
+
+
+## Proposal
+
+
+To achieve a two-node OpenShift cluster, we are leveraging traditional high-availability concepts and technologies. The proposed solution includes:
+
+1. Leverage of the Full RHEL-HA Stack:
+   * Run the RHEL-HA stack “under to kubelet” (directly on the hardware, not as an OpenShift workload)
+   * Corosync for super fast failure detection, membership calculations, which in turn will trigger Pacemaker to apply Fencing based on Corosync quorum/membership information.
+   * Pacemaker for integrating membership and quorum information, driving fencing, and managing if/when kubelet and etcd can be started
+   * Pacemaker models kubelet and cri-o as a clone (much like a ReplicaSet) and etcd as a “promotable clone” (think a construct designed for leader/follower style services).  Together with fencing and quorum, this ensures that an isolated node that reboots is inert and can do no harm.
+   * Pacemaker is [configured](//TODO mshitrit add link) to manage etcd/cri-o/kubelet, it will start/stop/restart those services using a script or an executable.
+   * Pacemaker does not understand what it is managing, and expects an executable or script that knows how to start/stop/monitor (and optionally promote/demote) the service.
+     1. Likely we would need to create one for etcd
+     2. For kubelet and cri-o we can likely use the existing systemd unit file
+2. Failure Scenarios:
+   * Implement detailed handling procedures for cold boots, network failures, node failures, kubelet failures, and etcd failures using the RHEL-HA stack.
+    [see examples](#failure-handling)
+3. Fencing Methods:
+   * We plan to use Baseboard Management Controller (BMC) as our primary fencing method, the premise of using BMC for fencing is that a node that is powered off, or was previously powered off and configured to be inert until quorum forms, is not in a position to cause corruption or diverging datasets.  Sending power-off (or reboot) commands to the peer’s BMC achieves this goal.
+
+
+### Workflow Description
+
+#### Cluster Creator Role:
+* The Cluster Creator will automatically install the 2NO (by using an installer), installation process will include the following steps:  
+  * Deploys a two-node OpenShift cluster
```

Comment:
Should be clearer now.
I think the options are that a cluster is 2NO when:
- 2 nodes, and
- the proposed externallyManagedEtcd field is true, and
- (optional?) a new feature gate is enabled

Reply:
+1 to using a feature gate, but that only flags that the cluster is not upgradeable because it's using a tech preview or even dev preview feature. By the time it goes GA that will be promoted and the cluster will be upgradeable

Still useful to avoid changes breaking the main payload though